clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -240,7 +240,11 @@ makeBinaryAtomicValue(CIRGenFunction &cgf, mlir::cir::AtomicFetchKind kind,
 
   Address destAddr = checkAtomicAlignment(cgf, expr);
   auto &builder = cgf.getBuilder();
-  auto intType = builder.getSIntNTy(cgf.getContext().getTypeSize(typ));
+  mlir::cir::IntType intType;
+  if (expr->getArg(0)->getType()->getPointeeType()->isUnsignedIntegerType())
+    intType = builder.getUIntNTy(cgf.getContext().getTypeSize(typ));
+  else
+    intType = builder.getSIntNTy(cgf.getContext().getTypeSize(typ));
```

Comment:
You can also do a oneliner here:
```
mlir::cir::IntType intType = expr->getArg(0)->getType()->getPointeeType()->isUnsignedIntegerType() ? builder.getUIntNTy(cgf.getContext().getTypeSize(typ)) : builder.getSIntNTy(cgf.getContext().getTypeSize(typ));
```
clang-format will help out

Reply:
done!